front/pages/w/[wId]/builder/assistants/dust.tsx

```diff
@@ -0,0 +1,284 @@
+import { useRouter } from "next/router";
+import { useContext } from "react";
+
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { AppLayoutSimpleCloseTitle } from "@app/components/sparkle/AppLayoutTitle";
+import { subNavigationAdmin } from "@app/components/sparkle/navigation";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { Authenticator, getSession, getUserFromSession } from "@app/lib/auth";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { APIError } from "@app/lib/error";
+import { useAgentConfigurations, useDataSources } from "@app/lib/swr";
+import { AgentConfigurationType } from "@app/types/assistant/agent";
+import { DataSourceType } from "@app/types/data_source";
+import { UserType, WorkspaceType } from "@app/types/user";
+
+const { GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType;
+  owner: WorkspaceType;
+  gaTrackingId: string;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    context.params?.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner || !user || !auth.isBuilder()) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: {
+      user,
+      owner,
+      gaTrackingId: GA_TRACKING_ID,
+    },
+  };
+};
+
+export default function EditDustAssistant({
+  user,
+  owner,
+  gaTrackingId,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { agentConfigurations, mutateAgentConfigurations } =
+    useAgentConfigurations({
+      workspaceId: owner.sId,
+    });
+  const { dataSources, mutateDataSources } = useDataSources(owner);
+
+  const sortedDatasources = dataSources.sort((a, b) => {
+    if (a.connectorProvider && !b.connectorProvider) {
+      return -1;
+    }
+    if (!a.connectorProvider && b.connectorProvider) {
+      return 1;
+    }
+    return a.name.localeCompare(b.name);
+  });
+
+  const dustAgentConfiguration = agentConfigurations?.find(
+    (c) => c.name === "dust"
+  );
+  if (!dustAgentConfiguration) {
+    return null;
+  }
+
+  const handleToggleAgentStatus = async (agent: AgentConfigurationType) => {
+    if (agent.status === "disabled_free_workspace") {
+      sendNotification({
+        title: "Dust Assistant",
+        description: `@${agent.name} is only available on our paid plans. Contact us at team@dust.tt to get access.`,
+        type: "error",
+      });
+      return;
+    }
+    if (agent.status === "disabled_missing_datasource") {
+      sendNotification({
+        title: "Dust Assistant",
+        description: `@${agent.name} is not available because you have not configured any data sources.`,
```

Comment:
The Dust assistant requires at least one data source or connection to be enabled to operate.